devops/scripts/build_and_register_bundle.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+set -e
+
+function usage() {
+    cat <<USAGE
+
+    Usage: $0 [-u --tre_url]  [-c --current] [-i --insecure]
+
+    Options:
+        -r, --acr-name        Azure Container Registry Name
+        -t, --bundle-type     Bundle type, workspace or workspace_service
+        -c, --current:        Make this the currently deployed version of this template
+        -i, --insecure:       Bypass SSL certificate checks
+        -u, --tre_url:        URL for the TRE (required for automatic registration)
+        -a, --access-token    Azure access token to automatically post to the API (required for automatic registration)
+USAGE
+    exit 1
+}
+
+# if no arguments are provided, return usage function
+if [ $# -eq 0 ]; then
+    usage # run usage function
+fi
+
+current="false"
+
+while [ "$1" != "" ]; do
+    case $1 in
+    -u | --tre_url)
+        shift
+        tre_url=$1
+        ;;
+    -r | --acr-name)
+        shift
+        acr_name=$1
+        ;;
+    -t | --bundle-type)
+        shift
+        case $1 in
+        workspace)
+        ;;
+        workspace_service)
+        ;;
+        user_resource)
+        ;;
+        *)
+            echo "Bundle type must be workspace, workspace_service or user_resource, not $1"
+            exit 1
+        esac
+        BUNDLE_TYPE=$1
+        ;;
+    -c| --current)
+        current="true"
+        ;;
+    -i| --insecure)
+        insecure=1
+        ;;
+    -a | --access-token)
+        shift
+        access_token=$1
+        ;;
+    *)
+        usage
+        ;;
+    esac
+    shift # remove the current value for `$1` and use the next
+done
+
+if [[ -z ${BUNDLE_TYPE+x} ]]; then
+    echo -e "No bundle type provided\n"
+    usage
+fi
+
+# This script assumes that it is being run in the BUNDLE_DIR so we can take the
+# current working directory
+BUNDLE_DIR="$(pwd)"
+TEMPLATE_NAME=$(yq eval '.name' ${BUNDLE_DIR}/porter.yaml)
+
+case "${BUNDLE_TYPE}" in
+  ("workspace") TRE_GET_PATH="api/workspace-templates" ;;
+  ("workspace_service") TRE_GET_PATH="api/workspace-service-templates" ;;
+esac
+
+export TOKEN=$(curl -X POST -H "Content-Type: application/x-www-form-urlencoded" -d \
+  "grant_type=password&resource=${RESOURCE}&client_id=${CLIENT_ID}&username=${USERNAME}&password=${PASSWORD}&scope=default)" \
+  https://login.microsoftonline.com/${AUTH_TENANT_ID}/oauth2/token | jq -r '.access_token')
+
+# We now need to traverse back to our scripts directory
+# TODO: Fix this traversing
+cd ../../../
+make register-bundle DIR=${BUNDLE_DIR}
```

Comment:
Hmm, @ross-p-smith  is there a reason these stages need to be separate. Is the only difference one gets the token and one does not?

Reply:
So I've left the existing development flow of 
- `make porter-build`
- `make register-bundle`

so that developers can authorise themselves

There is a new target `make build-and-register-bundle: porter-build` which does it all and manages the token

Reply:
Ok, makes sense, do we need to add this into the docs somewhere, can see it causing confusion. What each command does and when to use? I know this stuff and makes my head hurt.

Happy for you to create a separate issue, then can get this merged?

Reply:
Deal - I've raised #1193 which I will do whilst I am waiting for builds